physical/raft/raft_autopilot.go

```diff
@@ -389,21 +389,31 @@ func (d *Delegate) KnownServers() map[raft.ServerID]*autopilot.Server {
 // goroutine.
 func (d *Delegate) RemoveFailedServer(server *autopilot.Server) {
 	go func() {
-		if _, ok := d.deadServerRemovals.Load(server.ID); ok {
+		success := false
+		d.l.Lock()
+		defer func() {
+			if success {
```

Comment:
Why only on success?  The goal here is to avoid proliferation of duplicate goroutines, so as long as we're returning I'd say we can remove from the map.

Reply:
We want to do it only on success. Autopilot library might call `RemoveFailedServer` for the same server ID after every reconcile interval until it is actually removed from the raft config. In that case, we want the already initiated removal to be there in the map.

Reply:
Say that we get a failure on our call to RemoveServer so we return and leave the entry in inflightRemovals.  Doesn't that mean that on the next reconcile interval, when autopilot asks us again to remove the server, we'll check the map, see there's an entry, and do nothing?

Reply:
Ah, right. Fixed.